sound/soc/codecs/rt712-sdca.h

```diff
@@ -191,6 +221,7 @@ struct  rt712_sdca_priv {
 enum {
 	RT712_AIF1,
 	RT712_AIF2,
+	RT712_AIF3,
```

Comment:
how can we make sure that AIF3 is never used with a VA chip?

I find the whole thing a bit complicated if the information is hidden the machine integration could be problematic, no?

also how would we select a topology that sometimes has a dmic dailink and sometimes not?

Reply:
It is better to have different ADR between -VA and -VB. Or we might construct topology name dynamically, something like https://github.com/thesofproject/linux/pull/4847/commits/0930b9424667422db3fc7695bb525e89fa530e9d. But to be honest, I think it will be painful to do it.